src/encoded/static/components/inputs/file.js

```diff
@@ -2,6 +2,39 @@ import React from 'react';
 import PropTypes from 'prop-types';
 
 
+/**
+ * Depending on the filename extension, adjust the type and base64-encoded contents from the
+ * defaults that FileReader assigns. Return unchanged results for any file types that don't need
+ * special treatment.
+ * @param {object} rFile Object from <input type="file" />
+ * @param {string} readerResult FileReader `result` property.
+ *
+ * @return {object} -
+ *   type - mime type for file
+ *   href - base64-encoded contents of file
+ */
+const adjustFileValues = (rFile, readerResult) => {
+    let type = rFile.type;
+    let href = readerResult;
+
+    // https://stackoverflow.com/questions/190852/how-can-i-get-file-extensions-with-javascript/1203361#answer-12900504
+    const extension = rFile.name.slice((Math.max(0, rFile.name.lastIndexOf('.')) || Infinity) + 1);
```

Comment:
You could, but the referenced StackOverflow thread’s participants had gone over different methods in detail and came up with a number of solutions. This one I picked not because it was fastest (it was in fact one of the slower ones) but it took care of the most odd cases, such as file names starting with a dot.